docs/decisions/0005-architecture-and-data-modeling.rst

```diff
@@ -0,0 +1,149 @@
+- All modifications to the Authorization Engine configuration (model, adapters, etc.) should be done through the Open edX Layer, so no forks of Casbin are needed.
+
+Interact with the Policy Store via the Open edX Layer
+------------------------------------------------------
+- The policy store, which can be a database or configuration files (e.g., ``authz.policy``) will be accessed and managed through the Open edX Layer. No direct access to the policy store should be made by services.
+- The Open edX Layer will handle loading policies from the policy store into the Authorization Engine and ensure that policies are kept up to date.
+- The Open edX Layer will also manage the separation between static policies (shipped with services) and dynamic policies (managed via the policy data store) to ensure clarity and maintainability.
+
+Clients Interact Only via the Open edX Layer
+--------------------------------------------
+- Services and other clients (e.g., MFEs) will interact with the authorization system exclusively through the Open edX Layer's Enforcement API.
+- Services will not implement their own authorization logic or interact directly with Casbin internals.
+- Services can request authorization decisions by calling the Enforcement API with the necessary context (e.g., user, action, resource) with optional context and receiving a decision (allow/deny) in response.
+
+#. Data & Storage Model
+========================
+
+Use a MySQL as the Main Backend for the Policy Store
+----------------------------------------------------
+- Use MySQL as the main backend for persistent storage for policies (policy store), leveraging Casbin's Django ORM adapter for integration and our own modifications to the adapter as needed in our own ``engine/``` package.
+- Use the same schema used by Casbin's Django ORM adapter, which includes a table for storing policies with:
+  - ``id``: A unique identifier for each policy. This depends on the database backend. In our case, MySQL uses an auto-incrementing integer.
+  - ``ptype``: The policy type (e.g., ``p`` for policy, ``g, g2`` for grouping).
+  - ``v0, v1, v2, v3, v4, v5``: The policy fields that define the subject, action, object, and context. The exact meaning of these fields depends on the policy type and model configuration.
+- Optionally, include additional metadata fields in the policy table to support auditing and versioning.
+
+Store All Policies in the Policy Store
+--------------------------------------
+- All policies (i.e., any type of rule) should be stored in the policy store to ensure a single source of truth for authorization.
+- Use the policy store to manage RBAC mappings, such as user-role and role-permission assignments, using Casbin's grouping policies (``g, g2``).
+- Use Casbin's adapter APIs based on Django APIs to load policies from the policy store into the Authorization Engine at startup and whenever policies are updated.
```

Comment:
I assume this is "whenever dynamic policies are updated" and we're not putting a filesystem watcher on the policy file(s) or anything like that?